### PR TITLE
Refine project and Add AutoLogin

### LINKED ISSUE #2: Improve code style
- Remove comments that contain unused code.
- Do not use underscore variable name, maybe you should fix the eslint, or use a better base like the airbnb one.
- Replace function binding in constructor with arrow function.
```
# current one
onLogin() { ... }

# with arrow function
onLogin = () => { ... }
```
- Consider using a prettier to make the code looks even better, but optional IMO, see https://github.com/prettier/prettier.

### LINKED ISSUE #3: Fix login response handling
The current one is:
```js
onLogin() {
    skygear.auth.loginOAuthProviderWithPopup('google').then(
      // sign in was successful
      user => {
        console.info('Login success', user);
        console.info('Access token:', skygear.auth.accessToken);
        console.info('Username:', skygear.auth.currentUser.username);

        // status becomes signed in
        this.setState({ status: this.statuses[1] });
        skygear.auth.whoami().then(
          // whoami returns a Record type
          record => this.setState({ user: JSON.stringify(record.toJSON(), null, 2) }),
          this.onError('')
        );
        skygear.auth.getOAuthProviderProfiles().then(
          // getOAuthProviderProfiles return a JSON type
          profileJson => this.setState({ profile: JSON.stringify(profileJson, null, 2) }),
          this.onError('')
        );
      },
      // sign in was unsuccessful
      this.onError('Login failure')
    );
  }
```

1. The handling of `skygear.auth.loginOAuthProviderWithPopup('google')` looks wrong. The whole login operation should NOT be considered success until all checking has been done.
1. `skygear.auth.whoami()` is redundant, it reflects the same response of `skygear.auth.loginXXX`.

### LINKED ISSUE #4: Remove `whoami` call in logout
Like #3.

---

A side note, the current one:
```js

  onLogout() {
    skygear.auth.logout().then(
      this.isCurrentUserNull() ?
        this.onError('Logout failure') :
        user => {
          console.info('Logout success');
          this.setState(this.default_states);
        },
      this.onError('Logout failure')
    );
  }
```

The above code has two problems.

1. `this.isCurrentUserNull()` executes immediately when `onLogout` is called.
1. The result of `this.isCurrentUserNull()` is never checked.

A proper flow should be:

```js
skygear.auth.logout().then(() => {
  return this.isCurrentUserNull();
}).then(() => {
  console.info('Logout success');
  this.setState(this.default_states);
}).catch(this.onError('Logout failure'));
```

### LINKED ISSUE #5: Auto login if not logout last time
I think it would be good to demonstrate how to implement auto login, since this is required in most application.

### Auto login flow

1. In `componentDidMount` of `App`, check `skygear.auth.accessToken`.
1. If access token exists, call `skygear.auth.whoami()` and `skygear.auth.getOAuthProviderProfiles()`.
1. If access token does not exist, just display `not yet signed in` like the current one.
1. If both `skygear.auth.whoami()` and `skygear.auth.getOAuthProviderProfiles()` returns successful results, display user and profile data, the displayed information should look the same as that when a user successfully logged in with the `Login with Google` button.
1. If either of the call `skygear.auth.whoami()` or `skygear.auth.getOAuthProviderProfiles()` get error, call `skygear.auth.logout()` for the user.

### Code needs to be changed

With the current implementation, if you call `skygear.auth.accessToken` in `componentDidMount` of `App`, it would always return `null`.

The reason is `skygear.config` in `index.js` is asynchronous, the easiest fix is to render the react app AFTER `skygear.config` is finished, in this way you can always assume `skygear` is ready to use in your components.

### PR DESCRIPTION
**Pull request by bryanchun**
Resolve #2 
Resolve #3 
Resolve #4 
Resolve #5 
If this PR could close all the issues, I can start to update the tutorial according to the refined code.

Please note that when I wanted to use `Promise.all` to confirm that all promises are resolved so that I can group all the `then` and `catch` actions into one block, this worked for `componentDidMount` but failed at `this.onLogin`. Here, the promises must be run sequentially but there is no easily understandable vanilla code that does the same as `Promise.series` to synchronise the calls, that's why I resorted to nested promises.

connect #2

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -5,110 +5,84 @@ class App extends Component {
...
+            }).catch(this.onError('Login failure at getOAuthProviderProfiles'));  
+        }).then(() => {
+          // sign in was successful
+          console.info('Login success');
+          console.info('Access token:', skygear.auth.accessToken);
+          console.info('Username:', skygear.auth.currentUser.username);
+          // status becomes signed in
+          this.setState({ status: this.statuses[1] });
+        }).catch(this.onError('Login failure at loginOAuthProviderWithPopup'));
+    };
```
Comment (Steven-Chan) [inline src/App.js]: You can run promises in series with:

```js
skygear.auth.loginOAuthProviderWithPopup(xxx)
  .then((xxx) => {
     // xxxx
    return skygear.auth. getOAuthProviderProfiles();
  }).then((xxx) => {
    // xxx
  })
  .catch((xxx) => {
    // xxx;
  });
```

Comment (Steven-Chan) [reply src/App.js]: If you don't return a promise in the `then` function, the next `then` function would be executed immediately.

So your updated version runs into the same problem of the old code.

**Review by Steven-Chan — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (bryanchun) [reply src/App.js]: Thank you for the correction. I am not yet familiar with promises and the reason I tried to use nested promises (with hesitation) was I didn't realise I can get the resolved `value` of the second promise in the second `then`'s callback parameter. Now I know, and I will change accordingly.

**Review by bryanchun — COMMENTED** [review]
request for review

**Review by Steven-Chan — APPROVED** [review]

**Pull request merged**